org-roam-bibtex.el

```diff
@@ -8,9 +8,9 @@
 ;; URL: https://github.com/org-roam/org-roam-bibtex
 ;; Keywords: bib, hypermedia, outlines, wp
 ;; Version: 0.6.0
-;; Package-Requires: ((emacs "27.2") (org-roam "2.0.0") (bibtex-completion "2.0.0") (org-ref "1.1.1"))
+;; Package-Requires: ((emacs "27.2") (org-roam "2.0.0") (bibtex-completion "2.0.0") (org-mode "9.5"))
 
-;; Soft dependencies: projectile, persp-mode, helm, ivy, hydra
```

Comment:
Helm, Ivy and Hydra should stay mentioned as soft dependencies. These packages are required if a user decides to choose non-default `orb-insert` and `orb-note-actions` interfaces. They are not required to successfully install and use the package with default values. Org-ref may be moved onto this list.

Reply:
Done.

Note that `bibtex-actions-read` provides a "generic" completing-read-multiple interface. 

That's what I use for my `org-cite` "insert processor".

Reply:
BUT ... hang on. With org-cite, you can remove this function, and just use `org-cite-insert`. Then people can plugin their preferred insert processor like so:

```elisp
(setq org-cite-insert-processor 'foobar)
```

... and it will work anywhere in Org.

Reply:
`orb-insert-link` is a little bit more than just inserting a citation. By default, it inserts an ID link to an Org-roam node associated with the BibTeX entry chosen from the interface, or runs `org-roam-capture` if the note does not exist yet: [[Title][id:note-id]]. So basically it behaves like `org-roam-node-insert` but the completion candidates are populated from the BibTeX bibliography rather than from the Org-roam database. It can also insert links of type [[citation-key][id:note-id]] and Org-ref `cite:links` - by either running the command with an appropriate numerical prefix or by changing the default value of `orb-insert-link-description`. 

So this command is not a replacement for the insert functionalities from Org-ref, Bibtex-actions or Helm-bibtex, although it can be used to insert citation links. Dropping it completely is therefore not an option. Perhaps the same can be achieved by plugging in `orb-insert-link` logic into a custom `org-cite-insert` function, but ORB is not Org-cite. It is not a tool for citations, it is an add-on to Org-roam. It focuses on Org-roam things - notes. I think it would be confusing to call `org-cite-insert` to insert [[Title][id:note-id]] links.

That said, there is definitely a room for improvement. First of all, `orb-insert-link` should support `org-cite` citations. So, the list of valid values of `orb-insert-link-description` should be extended to `org-cite`. Also, the default value of `orb-insert-interface` is `generic`, which is a simple `completing-read` wrapper. It is rather unsophisticated and can be improved to play nicely with `completing-read` packages like `bibtex-actions`, `embark` and friends. Currently, it only retrieves citation keys and titles, If I remember correctly, so it may be modified to return a richer set fields.

Then again, maybe I'm using an incorrect wording but a "soft dependency" here means that some non-default settings require additional, optional packages. For example, if you set `orb-insert-interface` to `ivy-bibtex`, you should be have Ivy-bibtex installed, because you won't get the requested Ivy interface otherwise. The command, however, will not fail. It will just complain and fall back to the default `generic` interface. The same behaviour is implemented for other parts that require "soft dependencies". The things will work without the "soft dependencies" but will not deliver the sought non-default cosmetic tweaks. ORB does not enforce installation of "soft dependencies". Maybe I should update the docs and reformulate the phrase into something like "optional dependencies" or "interaction with other packages".

Reply:
Got it; you're right 

We do need to keep the distinction clear between citations and notes about citations.